src/CMakeLists.txt

```diff
@@ -7,4 +7,5 @@ target_include_directories(
         $<$<STREQUAL:$ENV{ARCH_NAME},wormhole_b0>:${CMAKE_CURRENT_SOURCE_DIR}/firmware/riscv/wormhole>
         $<$<STREQUAL:$ENV{ARCH_NAME},grayskull>:${CMAKE_CURRENT_SOURCE_DIR}/firmware/riscv/grayskull>
         $<$<STREQUAL:$ENV{ARCH_NAME},blackhole>:${CMAKE_CURRENT_SOURCE_DIR}/firmware/riscv/blackhole>
+        firmware/riscv
```

Comment:
shouldn't this be ${CMAKE_CURRENT_SOURCE_DIR}/firmware/riscv ?

And second question, why is this needed, aren't these headers already included with previous 3 lines? Or are previous 3 lines not needed now?
Do you maybe have a problem that in umd::device which is not conditionally compiled based on ARCH_NAME, you need to "#include blackhole/something.h", but umd::tests which is conditionally compiled based on ARCH_NAME will have "#include something.h" ?
It is probably not that simple to just change all "#include something.h" to ""#include arch/something.h"? Probably if there is some common file in all tests which should include based on arch...

Again, please just trying to clarify, no action for this PR, let's just leave info in the issue if there is anything additional to add https://github.com/tenstorrent/tt-umd/issues/270

Reply:
The implementations are always compiled:
https://github.com/tenstorrent/tt-umd/blob/main/device/CMakeLists.txt#L16-L18

And each one needs to know its piece:
https://github.com/tenstorrent/tt-umd/blob/main/device/grayskull/grayskull_implementation.cpp#L8-L9

The lines above seem to be abstracting away _which_ arch it is.  The effort to remove ARCH_NAME envvar would need to change such instances.